### PR TITLE
Fix #143

### LINKED ISSUE #143: Task tree: Double click to rename issue
Currently it is possible to rename one task in the tree by double clicking on it. This breaks the naming convention e.g. it is possible to rename `test(exopy_hqc_legacy.ApplyMagFieldTask)` into `test`. This might lead to other issues.
This issue doesn't exist when I use right-click->rename.

Solution: Double click to rename should be disabled or fixed to behave light right-click->rename.

### PR DESCRIPTION
**Pull request by rassouly**
Disable double-click to rename.

Use commitData signal instead of itemChanged to fix a bug where the
class name would disappear after clicking rename then hitting enter.

Newline at the end of line fix.

Should close #143.

Some code of my other PR #142 is also here because it fixes a bug related to renaming and can be merged independently of #142.

### CODE REVIEW

**Comment by MatthieuDartiailh** [comment]
Can you rebase on master to fix the conflict ?

File: exopy/utils/widgets/qt_tree_widget.py [hunk 1]
```diff
@@ -766,17 +769,13 @@ def _drop_object(self, node, obj, dropped_object, make_copy=True):
...
-        # The node data may not have been set up for the nid yet.  Ignore it if
-        # it hasn't.
-        try:
-            _, node, obj = self._get_node_data(nid)
-        except Exception:
-            return
-
+        nid = self.get_widget().currentItem()
+        col = self.get_widget().currentColumn()
+        _, node, obj = self._get_node_data(nid)
```
Comment (MatthieuDartiailh) [inline exopy/utils/widgets/qt_tree_widget.py]: Are you sure the try except is not needed anymore ?

**Comment by MatthieuDartiailh** [comment]
Thanks for rebasing.

File: exopy/utils/widgets/qt_tree_widget.py (hunk 1 above)
Comment (rassouly) [reply exopy/utils/widgets/qt_tree_widget.py]: I've tested it for a few minutes and it worked.
Moreover, `commitData` is a lot more selective that `itemChanged` and is only called when the user finishes editing an item so I don't think this signal can be emitted before the data has been set up.

Comment (MatthieuDartiailh) [reply exopy/utils/widgets/qt_tree_widget.py]: Ok. I will merge tomorrow then *or later tonight).

**Pull request merged**